core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/status/AmbariClusterStatusUpdater.java

```diff
@@ -43,62 +35,43 @@
-            TLSClientConfig clientConfig = tlsSecurityService.buildTLSClientConfig(stack.getId(), cluster.getAmbariIp());
-            AmbariClusterStatus clusterStatus = clusterStatusFactory.createClusterStatus(ambariClientProvider.getAmbariClient(
-                    clientConfig, cluster.getUserName(), cluster.getPassword()), blueprintName);
-            updateClusterStatus(stack, clusterStatus);
-        }
-    }
-
-    private boolean isClusterStatusCheckNecessary(Stack stack) {
-        boolean result = false;
-        Cluster cluster = stack.getCluster();
-        if (stack.isDeleteCompleted() && cluster != null) {
-            result = failedStatuses.contains(stack.getStatus()) || failedStatuses.contains(cluster.getStatus())
-                    || (stack.isAvailable() && (cluster.isAvailable() || cluster.isStopped()));
-        }
-        return result;
+    public void updateClusterStatus(Stack stack, Cluster cluster) throws CloudbreakSecuritySetupException {
+        Long stackId = stack.getId();
+        String blueprintName = cluster != null ? cluster.getBlueprint().getBlueprintName() : null;
+        TLSClientConfig clientConfig = tlsSecurityService.buildTLSClientConfig(stackId, cluster.getAmbariIp());
+        AmbariClusterStatus clusterStatus = clusterStatusFactory.createClusterStatus(ambariClientProvider.getAmbariClient(
+                clientConfig, cluster.getUserName(), cluster.getPassword()), blueprintName);
+        updateClusterStatus(stackId, stack.getStatus(), cluster, clusterStatus);
     }
 
-    private boolean updateClusterStatus(Stack stack, AmbariClusterStatus ambariClusterStatus) {
-        boolean result = false;
+    private void updateClusterStatus(Long stackId, Status stackStatus, Cluster cluster, AmbariClusterStatus ambariClusterStatus) {
+        Status statusInEvent = stackStatus;
+        String statusReason;
         if (ambariClusterStatus != null && isUpdateEnabled(ambariClusterStatus.getStatus())) {
-            boolean clusterUpdated = updateClusterStatus(stack, ambariClusterStatus.getClusterStatus());
-            boolean stackUpdated = updateStackStatus(stack, ambariClusterStatus.getStackStatus(), ambariClusterStatus.getStatusReason());
-            if (clusterUpdated && !stackUpdated) {
-                cloudbreakEventService.fireCloudbreakEvent(stack.getId(), ambariClusterStatus.getStackStatus().name(), ambariClusterStatus.getStatusReason());
+            if (updateClusterStatus(stackId, cluster, ambariClusterStatus.getClusterStatus())) {
+                statusInEvent = ambariClusterStatus.getStackStatus();
+                statusReason = ambariClusterStatus.getStatusReason();
+            } else {
+                statusReason = "The cluster's state is up to date.";
             }
-            result = stackUpdated || clusterUpdated;
+        } else {
+            statusReason = "The cluster's state is ambiguous, stay unchanged.";
```

Comment:
please change this message to something more user friendly like "There are stopped and running Ambari services as well. Restart or stop all of them and try syncing later."